src/commands/external.rs

```diff
@@ -49,10 +50,20 @@ pub async fn execute_external_subcommand(
             }
         }
         Err(Error::NotFound(e)) => {
-            tracing::debug!("Tried to resolve {plugin_name} to plugin, got {e}");
-            eprintln!("Error: '{plugin_name}' is not a known Spin command. See spin --help.\n");
-            print_similar_commands(app, &plugin_name);
-            process::exit(2);
+            if plugin_name == "cloud" {
+                println!("The `cloud` plugin is required. Installing now.");
+                let plugin_installer = Install {
+                    name: Some("cloud".to_string()),
+                    yes_to_all: true,
+                    ..Default::default()
+                };
+                plugin_installer.run().await?;
```

Comment:
Do we want to attempt re-calling `spin cloud deploy` after installing the plugin? I can't tell if the newly installed plugin gets invoked after answering "yes" to install the plugin.

Reply:
yes instead of returning, currently it will proceed with the command execution after install, so the flow looks like the following:

```sh
$ spin deploy
The `cloud` plugin is required. Installing now.
Plugin 'cloud' was installed successfully!
Uploading cloud_start version 0.1.0+XXXXXXXX...
Deploying...
Waiting for application to become ready... ready
Available Routes:
  cloud-start: https://cloud-start-xxxxxxxx.fermyon.app/ (wildcard)
```